internal/ast/ast.go

```diff
@@ -299,6 +307,23 @@ type (
 		Position
 		Label Stmt // Break statement label
 	}
+
+	// ForStatement node represents a for statement
+	ForStatement struct {
+		Position
+		VarDecl   Stmt       // For initializer
+		Cond      Stmt       // Condition needed to match
+		Increment Expr       // For Increment expression
+		Body      *BlockStmt // For body
+	}
+
+	// ForInStatement node represents a for in statement
+	ForInStatement struct {
+		Position
+		Left  *Ident     // Value on the left side of for in containing the value of the current iteration
+		Right *Ident     // Value on the right side of for in containing an array to iterate over
```

Comment:
[According to tree-sitter grammar](https://github.com/tree-sitter/tree-sitter-javascript/blob/master/grammar.js#L290) `left` and `right` nodes could be either a ident or a destructuring_pattern, but we don't support this node yet, so I think that would be better change this fields to be a `Expr`.